Sources/SwiftSyntaxParser/SyntaxParser.swift

```diff
@@ -86,18 +86,12 @@ public enum SyntaxParser {
     var utf8Source = source
     utf8Source.makeContiguousUTF8()
 
-    let rawSyntax = parseRaw(source: utf8Source,
-                             parseTransition: parseTransition,
-                             filenameForDiagnostics: filenameForDiagnostics,
-                             languageVersion: languageVersion,
-                             enableBareSlashRegexLiteral: enableBareSlashRegexLiteral,
-                             diagnosticHandler: diagnosticHandler)
-
-    let base = _SyntaxParserInterop.nodeFromRetainedOpaqueRawSyntax(rawSyntax)
-    guard let file = base.as(SourceFileSyntax.self) else {
-      throw ParserError.invalidSyntaxData
-    }
-    return file
+    return try parseImpl(source: utf8Source,
```

Comment:
I just renamed `parseRaw` to `parseImpl` and now it's returning `SourceFileSyntax` instead of `CClientNode`. Because we need to create `SourceFileSyntax` while the `SyntaxArena` is alive.

Reply:
Oh, now I see that `parseImpl` is quite a large function. I somehow thought it was 10-ish lines. Probably mixed it up with the other `parse` function below. Makes sense to keep it as-is. Sorry.